src/ArmTemplates/Common/Templates/Abstractions/TemplateParameterProperties.cs

```diff
@@ -1,11 +1,22 @@
-﻿namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Abstractions
+﻿using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Commands.Applications;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Abstractions
 {
     public class TemplateParameterProperties
     {
-        public string type { get; set; }
-        public TemplateParameterMetadata metadata { get; set; }
-        public string[] allowedValues { get; set; }
-        public string defaultValue { get; set; }
-        public string value { get; set; }
+        public string Type { get; set; }
+        public TemplateParameterMetadata Metadata { get; set; }
+        public string[] AllowedValues { get; set; }
+        public string DefaultValue { get; set; }
+        public string Value { get; set; }
+
+        public static TemplateParameterProperties Create(string metadataDescription, string type) => new()
```

Comment:
Why a static Create method instead of a constructor with these two params?

Reply:
How would it affect other instantiations? 

It would be a **new** constructor, so it wouldn't be a breaking change for anything. And we can always also have a ctor with no params in case it's currently used anywhere else.

Reply:
Sorry, thought about other thing. Changed it to constructor and a public one